src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -0,0 +1,1021 @@
+        //namespace Preview_Feature_Scratch
+        //{"+
+        //@"
+
+        //    class Program : IProgram
+        //    {
+        //        static void Main(string[] args)
+        //        {
+        //            new Program();
+        //        }
+        //    }
+
+        //    public interface IProgram
+        //    {
+        //        public static abstract bool [|StaticMethod|]();
+        //        public static abstract bool [|AProperty|] { [|get|]; }
+        //    }
+        //}
+
+        //    ";
+        //            await new VerifyCS.Test
+        //            {
+        //                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+        //                TestState =
+        //                {
+        //                    Sources =
+        //                    {
+        //                        csInput, requiresPreviewFeaturesAttribute
+        //                    }
+        //                },
+        //            }.RunAsync();
+        //        }
+
+        [Fact]
+        public async Task TestMarkedPreviewInterface()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program : IProgram
+    {
+        static void Main(string[] args)
+        {
+            new Program();
+        }
+
+        public void [|UnmarkedMethodInMarkedInterface|]() { }
+
+    }
+
+    [RequiresPreviewFeatures]
+    public interface IProgram
+    {
+        public void UnmarkedMethodInMarkedInterface() { }
+    }
+}
+
+    ";
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestState =
+                {
+                    Sources =
+                    {
+                        csInput, requiresPreviewFeaturesAttribute
+                    }
+                },
+                MarkupOptions = CodeAnalysis.Testing.MarkupOptions.UseFirstDescriptor
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestMarkedEmptyPreviewInterface()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class [|Program|] : IProgram
+    {
+        static void Main(string[] args)
+        {
+            new Program();
+        }
+    }
+
+    [RequiresPreviewFeatures]
+    public interface IProgram
+    {
+    }
+}
+
+    ";
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestState =
+                {
+                    Sources =
+                    {
+                        csInput, requiresPreviewFeaturesAttribute
+                    }
+                },
+                MarkupOptions = MarkupOptions.UseFirstDescriptor,
+            }.RunAsync();
```

Comment:
Probably check the Diagnostic and/or message for some of the tests as they have different Diagnostics on different scenarios

Reply:
Yup, that is on my todo. Do you mind if I do it in a follow up PR?

Reply:
🆗